draft-ietf-quic-transport.md

```diff
@@ -2014,6 +2015,27 @@ Final offset:
   data written on this stream by the RST_STREAM sender.
 
 
+
+
+## DISINTEREST Frame {#frame-disinterest}
+
+An endpoint may use a DISINTEREST frame (type=0x08) to communicate that incoming
+data is being discarded on receipt at application request.  This signals a peer
+to abruptly terminate transmission on a stream.  The frame is as follows:
+
+~~~
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                        Stream ID (32)                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+The fields are:
+
+* Stream ID: The 32-bit Stream ID of the stream being ignored.
+
```

Comment:
There originally was, and @martinthomson suggested removing it (see his review in April).  The point of this frame is that the sender's side isn't inherently closed, so any reasons can still be communicated at the application layer.  That's certainly the case with HTTP, where the most common reason is that the server has already generated a response.  I have a mild preference for leaving it out, since this isn't targeted at error cases, but I'm largely ambivalent.  I'd suggest taking that design discussion to the issue or list.

Including the offset doesn't really matter -- the sender of DISINTEREST has received all the data they needed, which implies the receiver has already successfully sent at least that much.  Why do they care what the offset of not-caring was?  RST_STREAM includes the offset because that's necessary to synchronize flow control.  Not so here.